Source/MLX/MLXArray.swift

```diff
@@ -160,35 +160,30 @@ public final class MLXArray {
     /// let value = array[1].item(Float.self)
     /// ```
     public func item<T: HasDType>(_ type: T.Type) -> T {
-        self.eval()
+        precondition(self.size == 1)
 
-        var array_ctx = self.ctx
-        var free = false
         if type.dtype != self.dtype {
-            array_ctx = mlx_astype(self.ctx, type.dtype.cmlxDtype, StreamOrDevice.default.ctx)
-            mlx_array_eval(array_ctx)
-            free = true
+            return self.asType(type).item(type)
         }
 
-        // can't do it inside the else as it will free at the end of the block
-        defer { if free { mlx_free(array_ctx) } }
+        self.eval()
```

Comment:
That diff is a bit messy, but here is the new code:

```swift
    public func item<T: HasDType>(_ type: T.Type) -> T {
        precondition(self.size == 1)

        if type.dtype != self.dtype {
            return self.asType(type).item(type)
        }

        self.eval()
```

That is consistent with the other readout methods (below).  It simplifies the data conversion path and all the memory management.